packages/core/test-helpers/core-test-helpers-model-versions/src/test_bed.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import { startElasticsearch } from './elasticsearch';
+import { prepareModelVersionTestKit } from './test_kit';
+import type { ModelVersionTestBed } from './types';
+
+/**
+ * Create a {@link ModelVersionTestBed} that can be used for model version integration testing.
+ *
+ * @example
+ * ```ts
+ * describe('myIntegrationTest', () => {
+ *   const testbed = createModelVersionTestBed();
+ *   let testkit: ModelVersionTestKit;
+ *
+ *   beforeAll(async () => {
+ *     await testbed.startES();
+ *   });
+ *
+ *   afterAll(async () => {
+ *     await testbed.stopES();
+ *   });
+ *
+ *   beforeEach(async () => {
+ *     testkit = await testbed.prepareTestKit({
+ *       savedObjectDefinitions: [{
+ *         definition: mySoTypeDefinition,
+ *         modelVersionBefore: 1,
+ *         modelVersionAfter: 2,
+ *       }]
+ *     })
+ *   });
+ *
+ *   afterEach(async () => {
+ *     if(testkit) {
+ *       await testkit.tearsDown();
+ *     }
+ *   });
+ *
+ *   it('can be used to test model version cohabitation', async () => {
+ *     // last registered version is `1`
+ *     const repositoryV1 = testkit.repositoryBefore;
+ *     // last registered version is `2`
+ *     const repositoryV2 = testkit.repositoryAfter;
+ *
+ *     // do something with the two repositories, e.g
+ *     await repositoryV1.create(someAttrs, { id });
```

Comment:
I remember having done something like this for testing bwc update.
If we did go with the test I updated/added, we should go back and use this implementation instead.